engine/dlib/src/dlib/profile/profile_remotery.cpp

```diff
@@ -129,12 +169,15 @@ namespace dmProfile
 
     void SetThreadName(const char* name)
     {
+        if (!IsInitialized())
+            return;
+        DM_SPINLOCK_SCOPED_LOCK(g_ProfilerLock);
```

Comment:
On startup, another thread wanted to set the profiler name, before the system was initialized.